File: leetcode/600/LC643findMaxAverage.go
func findMaxAverage(nums []int, k int) float64 {
	sum, max := 0, 0
	left, right := 0, k

	for i := 0; i < right; i++ {
		sum += nums[i]
	}
	max = sum

	for right < len(nums) {
		sum += nums[right]
		right += 1
		sum -= nums[left]
		left += 1
		if sum > max {
			max = sum
		}
	}

	return float64(max) / float64(k)
}

File: leetcode/600/LC643FindMaxAverage.java
public class LC643FindMaxAverage {
    public double findMaxAverage(int[] nums, int k) {
        int sum = 0, max = 0;
        int left = 0, right = k;
        for (int i = 0; i < right; i++) {
            sum += nums[i];
        }
        max = sum;

        while (right < nums.length) {
            sum += nums[right++];
            sum -= nums[left++];
            max = Math.max(max, sum);
        }
        return (double) max / k;
    }
}
